File: android/src/main/java/com/onegini/mobile/sdk/reactnative/exception/OneginiWrapperErrors.kt
package com.onegini.mobile.sdk.reactnative.exception

enum class OneginiWrapperErrors(val code: String, val message: String) {
    IDENTITY_PROVIDER_NOT_FOUND("8001", "Identity provider not found"),
    WRONG_CONFIG_MODEL("8002", "Provided config model parameters are wrong"),
    MOBILE_AUTH_OTP_IS_DISABLED("8003", "The Mobile auth Otp is disabled"),
    WRONG_PIN_ERROR("8004", "Wrong pin provided"),
    RESOURCE_CALL_ERROR("8005", "Resource call finished with error"),
    AUTHENTICATOR_DOES_NOT_EXIST("8006", "The Fingerprint authenticator does not exist"),
    PROFILE_DOES_NOT_EXIST("8007", "The profileId does not exist"),
    PARAMETERS_NOT_CORRECT("8009", "The parameters to the called function are not correct"),
    REGISTRATION_NOT_IN_PROGRESS("8010", "Registration is currently not in progress"),
    FINGERPRINT_IS_NOT_ENABLED("80011", "The fingerprint is not enabled in the configuration"),
}

File: ios/OneginiWrapperErrors.swift
enum WrapperError: LocalizedError {
    case identityProviderNotFound
    case profileDoesNotExist
    case authenticatorDoesNotExist
    case malformedUrl
    case parametersNotCorrect(description: String)
    case noProfileAuthenticated
    case registrationNotInProgress
    case mobileAuthNotInProgress
    case authenticationNotInProgress
    case pinCreationNotInProgress
    case actionNotAllowed(description: String)
    case authenticatorNotRegistered
    
    var errorDescription: String? {
        switch self {
        case .identityProviderNotFound:
            return "Identity provider not found"
        case .profileDoesNotExist:
            return "The profile does not exist"
        case .authenticatorDoesNotExist:
            return "The authenticator does not exist"
        case .malformedUrl:
            return "The supplied url is malformed"
        case .parametersNotCorrect(let description):
            return "The supplied parameters are not correct: " + description
        case .noProfileAuthenticated:
            return "No profile is currently authenticated"
        case .registrationNotInProgress:
            return "Registration is currently not in progress"
        case .mobileAuthNotInProgress:
            return "There is currently no mobile authentication in progress"
        case .authenticationNotInProgress:
            return "Authentication is currently not in progress"
        case .pinCreationNotInProgress:
            return "Pin creation is currently not in progress"
        case .actionNotAllowed(let description):
            return description
        case .authenticatorNotRegistered:
            return "The authenticator is not registered"
        }
    }
    var code: Int {
        switch self {
        case .identityProviderNotFound:
            return 8001
        case .profileDoesNotExist:
            return 8004
        case .authenticatorDoesNotExist:
            return 8006
        case .parametersNotCorrect:
            return 8009
        case .registrationNotInProgress:
            return 8010
        case .noProfileAuthenticated:
            return 8012
        case .mobileAuthNotInProgress:
            return 8013
        case .malformedUrl:
            return 8014
        case .authenticationNotInProgress:
            return 8015
        case .pinCreationNotInProgress:
            return 8016
        case .actionNotAllowed:
            return 8017
        case .authenticatorNotRegistered:
            return 8019
        }
    }
}
